src/instructlab/train/linux_train.py

```diff
@@ -235,14 +253,24 @@ def model_generate(user):
     )
 
     output_dir = "./training_results"
-    per_device_train_batch_size = 1
+    per_device_train_batch_size = determine_batch_size(device)
+    print(f"LINUX_TRAIN.PY: Training with batch size: {per_device_train_batch_size}")
+
+    use_bf16 = not use_fp16
 
+    if device.type == "cpu":
+        print("LINUX_TRAIN.PY: Disabling fp16 and bf16 for cpu training")
+        # CPU performance is very bad with fp16 or bf16, so disable both
+        use_fp16 = use_bf16 = False
```

Comment:
The best option to use likely depends on specific CPU used here. Xeon CPUs with AVX512 and ipex may actually get better performance leaving bf16 enabled. For the sake of better CPU defaults, this is probably fine. But this is another example where the work @derekhiggins is doing around exposing all of these training arguments would be useful so that users could ultimately tweak these for their specific system until or unless we get smart enough to make optimal decisions in most cases.

Reply:
I'll test this to compare on my HW but it will be early next week before I get a chance

Here is the PR I've been testing to allow arbitrary training options 
https://github.com/instructlab/instructlab/pull/1008

Reply:
Agreed. IMO, it'd be nice to have something like "profiles" and have it automatically select a reasonable default if none is specified.

Reply:
I have been facing a similar problem for Intel Gaudi (Habana Labs HPUs and SynpseAI) and created an abstract layer for Torch device properties. I just ripped the code out of my experimental branch, cleaned it up, and published it as draft PR #1015.